macros/generic_tests/recency.sql

```diff
@@ -4,7 +4,21 @@
 
 {% macro default__test_recency(model, field, datepart, interval, group_by_columns) %}
 
-{% set threshold = dbt.dateadd(datepart, interval * -1, current_timestamp_backcompat()) %}
+{#- Get columns in model #}
+{%- set columns = adapter.get_columns_in_relation(model) %}
+
+{#- Get column type of field in model #}
+{%- set column_type = columns | selectattr("name", "==", field) | map(attribute="data_type") | first %}
+
+{#- Use a date-based threshold if column type is DATE or datepart is day or greater #}
+{%- if column_type == 'DATE' and datepart in ['year', 'quarter', 'month', 'week', 'day'] %}
+  {%- set threshold = dbt.dateadd(datepart, interval * -1, dbt.safe_cast(dbt.current_timestamp_backcompat(), api.Column.translate_type("date"))) %}
+{%- else %}
+  {%- set threshold = dbt.dateadd(datepart, interval * -1, dbt.current_timestamp_backcompat()) %}
```

Comment:
There was a merge conflict which I had to resolve, and while I was there I changed these to `dbt.current_timestamp_backcompat()` which is the equivalent of what was `dbt_utils.current_timestamp()`.

Reply:
Got it!